rll-examples/disable-monitoring.sh

```diff
@@ -0,0 +1,116 @@
+      echo "RightLink 10 built-in monitoring is enabled. Disabling it."
+      rsc rl10 $retry_flags update /rll/tss/control enable_monitoring=false
+    else
+      echo "RightLink 10 built-in monitoring is disabled."
+    fi
+  fi
+fi
+
+# RL 6 relied on collectd as a standalone service, which sends data directly to
+# RightScale "sketchy" servers using the network plugin (UDP). Only the collectd
+# v4 format is understood by UDP-based infrastructure. RL10 uses the native OS
+# collectd, which can be collectd 4 or 5. In either case the data is sent over
+# http to the RightLink client which adds an auth header and sends it onto the
+# RightScale TSS servers. 
+# Ubuntu location: /etc/collectd/plugins, RHEL/CentOS location: /etc/collectd.d
+plugins_dir=/etc/collectd.d
+if [[ -d /etc/collectd/plugins ]]; then
+  plugins_dir=/etc/collectd/plugins
+fi
+collectd_conf=/etc/collectd.conf
+if [[ -e /etc/collectd/collectd.conf ]]; then 
+  collectd_conf=/etc/collectd/collectd.conf
+fi
+if [[ -e $collectd_conf ]]; then
+  echo "Collectd based monitoring detected."
+  locations=($plugins_dir/network.conf $plugins_dir/write_http.conf)
+  for filename in ${locations[*]}; do
+    if $sudo test -f "$filename"; then
+      if $sudo grep -E 'rightscale|rll/tss' "$filename" 2>/dev/null; then
+        echo "RightScale collectd-based monitoring is enabled. Disabling collectd configuration $filename."
+        backup_time=$(date -u +%Y%m%d%H%M%S)
+        $sudo mv "${filename}" "${filename}.${backup_time}"
+        restart_collectd=1
+      fi
+    fi
+  done
+
+  # From https://collectd.org/wiki/index.php/Table_of_Plugins, all known "write"
+  # plugins. If there are no write plugins enabled, collectd will spew errors
+  # continually and is non-functional. Disable it in that case.
+  write_plugins_regex="Plugin .*(amqp|carbon_writer|csv|network|rrdcached|rrdtool|unixsock|write_.*)"
+  if $sudo grep -E "$write_plugins_regex" $(ls $plugins_dir/*.conf $collectd_conf); then
+    echo "Found alternative collectd write plugin. Collectd will not be disabled."
+  else
+    echo "Found no collectd writer plugin other than the RightScale one."
+    echo "Disabling collectd service as it does not appear to be setup for anything other than RightScale monitoring."
+    disable_collectd=1
+  fi
+
+  if [[ "$disable_collectd" == "1" ]]; then
+    if which chkconfig; then
+      $sudo chkconfig collectd off
+    elif which update-rc.d; then
+      $sudo update-rc.d collectd disable
+    fi
+    echo "Collectd disabled. Stopping collectd service"
+    if $sudo service collectd status; then
+      $sudo service collectd stop
+    fi
+  elif [[ "$restart_collectd" == "1" ]]; then
+    echo "Collectd configuration changed. Restarting collectd service."
+    if $sudo service collectd status; then
+      $sudo service collectd restart
+    fi
+  else
+    echo "No changes to collectd configuration were needed."
+  fi
+fi
+
+if which rightlink >/dev/null 2>&1; then
```

Comment:
We do this `which rightlink >/dev/null` 3x - can we do it once and set it's existence to a variable - same way we do it with `sudo` and `retry_flags`?